src/sipm_simple_calibration.jl

```diff
@@ -0,0 +1,215 @@
+"""
+    sipm_simple_calibration(pe_uncal::Array)
+
+
+Perform a simple calibration for the uncalibrated p.e. spectrum array `pe_uncal`
+using just the 1 p.e. and 2 p.e. peak positions estimated by a peakfinder.
+
+Returns 
+    * `pe_simple_cal`: array of the calibrated pe array with the simple calibration
+    * `func`: function to use for the calibration
+    * `c`: calibration factor
+    * `peakpos`: 1 p.e. and 2 p.e. peak positions in uncalibrated amplitude
+    * `h_uncal`: histogram of the uncalibrated pe array
+    * `h_calsimple`: histogram of the calibrated pe array with the simple calibration
+"""
+
+function sipm_simple_calibration end
+export sipm_simple_calibration
+
+function sipm_simple_calibration(pe_uncal::Vector{<:Real}; 
+                                expect_noise_peak::Bool=false, 
+                                kwargs...)
+    
+    if expect_noise_peak
+        h_uncal, peakpos = find_peaks_noise_peak_exists(pe_uncal; kwargs...)
+    else
+        h_uncal, peakpos = find_peaks(pe_uncal; kwargs...)
+    end
+
+    # simple calibration
+    gain = peakpos[2] - peakpos[1]
+    c = 1/gain
+    offset = - (peakpos[1] * c - 1) 
+
+    f_simple_calib = Base.Fix1(*, c)
+    pe_simple_cal = pe_uncal .* c .+ offset
+
+    h_calsimple = histogram(pe_simple_cal, bins=0.5:.01:4.5)
+
+    result = (
+        pe_simple_cal = pe_simple_cal,
+        func = f_simple_calib, 
+        c = c,
+        offset = offset
+    )
+    report = (
+        peakpos = peakpos,
+        h_uncal = h_uncal, 
+        h_calsimple = h_calsimple
+    )
+    return result, report
+end
+
+
+
+# search the 1 p.e. and 2 p.e. peak, noise peak exists
+function find_peaks_noise_peak_exists(
+    amps::Vector{<:Real}; initial_min_amp::Real=1.0, initial_max_quantile::Real=0.99, 
+    step_size_min_amp::Real=1.0, peakfinder_σ::Real=1.0, peakfinder_threshold::Real=10.0
```

Comment:
The `peakfinder_σ` is 1.0. This means the peakfinder expects peaks with width of 1 bin. Is this not too small?

Reply:
to what value would you set it?

Reply:
I think the default is 2.0 in [RadiationSpectra.jl](https://github.com/JuliaPhysics/RadiationSpectra.jl/blob/f1ea9f8c18a67e2e5edb51544ccfca3f05401e15/src/PeakFinder/PeakFinder.jl#L371-L390)

Reply:
Maybe try it out, and then we can see